src/common/io/src/buffer/buffer_read_datetime_ext.rs

```diff
@@ -61,48 +70,179 @@ where R: BufferRead
+                let mut ts = dt.timestamp_micros();
+                ts += offset;
+                // TODO: need support timestamp_micros in chrono-0.4.22/src/offset/mod.rs
+                // use like tz.timestamp_nanos()
+                let (mut secs, mut micros) = (ts / 1_000_000, ts % 1_000_000);
+                if ts < 0 {
+                    secs -= 1;
+                    micros += 1_000_000;
+                }
+                Ok(tz.timestamp_opt(secs, (micros as u32) * 1000).unwrap())
+            };
+            if self.ignore(|b| b == b'z' || b == b'Z')? {
+                // ISO 8601 The Z on the end means UTC (that is, an offset-from-UTC of zero hours-minutes-seconds).
+                if dt.year() < 1000 {
+                    Ok(tz.from_utc_datetime(&NaiveDate::from_ymd(1000, 1, 1).and_hms(0, 0, 0)))
+                } else {
+                    let current_tz = dt.offset().fix().local_minus_utc();
+                    calc_offset(current_tz.into(), 0, &dt)
+                }
+            } else if self.ignore_byte(b'+')? {
+                self.parse_time_offset(tz, &mut buf, &dt, false, calc_offset)
+            } else if self.ignore_byte(b'-')? {
+                self.parse_time_offset(tz, &mut buf, &dt, true, calc_offset)
             } else {
-                res
+                // only datetime part
+                less_1000(dt)
             }
         } else {
-            res
-        };
-        // convert timestamp less than `1000-01-01 00:00:00` to `1000-01-01 00:00:00`
-        if dt.year() < 1000 {
-            Ok(tz.from_utc_datetime(&NaiveDate::from_ymd(1000, 1, 1).and_hms(0, 0, 0)))
+            // only date part
+            less_1000(dt)
+        }
+    }
+
+    fn parse_time_offset(
+        &mut self,
+        tz: &Tz,
+        buf: &mut Vec<u8>,
+        dt: &DateTime<Tz>,
+        west_tz: bool,
+        calc_offset: impl Fn(i64, i64, &DateTime<Tz>) -> Result<DateTime<Tz>>,
+    ) -> Result<DateTime<Tz>> {
+        if self.keep_read(buf, |f| (b'0'..=b'9').contains(&f))? != 2 {
+            return Err(ErrorCode::BadBytes(
+                "err with parse timezone, format like this:[+08:00]",
+            ));
+        }
+        let hour_offset: i32 = lexical_core::FromLexical::from_lexical(buf.as_slice()).unwrap();
+        if (0..15).contains(&hour_offset) {
+            buf.clear();
+            self.ignore_byte(b':')?;
+            if self.keep_read(buf, |f| (b'0'..=b'9').contains(&f))? != 2 {
+                return Err(ErrorCode::BadBytes(
+                    "err with parse timezone, format like this:[+08:00]",
+                ));
+            }
+            let minute_offset: i32 =
+                lexical_core::FromLexical::from_lexical(buf.as_slice()).unwrap();
+            // max utc: 14:00, min utc: 00:00
+            if (hour_offset == 14 && minute_offset == 0)
+                || ((0..60).contains(&minute_offset) && hour_offset < 14)
+            {
+                if dt.year() < 1970 {
```

Comment:
Can't there be an offset before 1970? The minimum supported value for `timestamp` is 1000

Reply:
Oh, It can be. Forget to modify it.